File: java/solutions/01-beginner/ex032_inheritance_override/InheritanceOverride.java
package fewolearning.exercises.beginner.ex032_inheritance_override;

/*
Exercise 032 - Inheritance override (reference solution).
*/
public final class InheritanceOverride {
    private InheritanceOverride() {
    }

    public static class Animal {
        public String describe() {
            return "an animal";
        }
    }

    public static class Dog extends Animal {
        @Override
        public String describe() {
            return super.describe() + " (a dog)";
        }
    }
}

File: kotlin/solutions/01-beginner/ex024_inheritance_override/InheritanceOverride.kt
package fewolearning.exercises.beginner.ex024_inheritance_override

/*
Exercise 024 - Inheritance override (reference solution).
*/
open class Animal {
    open fun describe(): String = "an animal"
}

class Dog : Animal() {
    override fun describe(): String = "${super.describe()}, specifically a dog"
}
